tests/integration/test_load_and_serve_agent.py

```diff
@@ -0,0 +1,134 @@
+
+from any_agent import AgentConfig, AgentFramework, AnyAgent, TracingConfig
+from any_agent.config import MCPStdio, ServingConfig
+from any_agent.tracing.trace import AgentTrace, _is_tracing_supported
+
+from queue import Queue
+
+from common.client import A2AClient
+from common.types import TaskSendParams, Message, TextPart
+
+FORMAT = "%(message)s"
+logging.basicConfig(
+    format=FORMAT,
+    datefmt="[%X]",
+)
+logger = logging.getLogger("any_agent_test")
+logger.setLevel(logging.DEBUG)
+
+def check_uvx_installed() -> bool:
+    """The integration tests requires uvx"""
+    try:
+        result = subprocess.run(  # noqa: S603
+            ["uvx", "--version"],  # noqa: S607
+            capture_output=True,
+            check=True,
+        )
+        return True if result.returncode == 0 else False  # noqa: TRY300
+    except Exception:
+        return False
+
+
+@pytest.mark.skipif(
+    os.environ.get("ANY_AGENT_INTEGRATION_TESTS", "FALSE").upper() != "TRUE",
+    reason="Integration tests require `ANY_AGENT_INTEGRATION_TESTS=TRUE` env var",
+)
+@pytest.mark.asyncio
+async def test_load_and_serve_agent(agent_framework: AgentFramework, tmp_path: Path) -> None:
+    kwargs = {}
+
+    tmp_file = "tmp.txt"
+
+    if not check_uvx_installed():
+        msg = "uvx is not installed. Please install it to run this test."
+        raise RuntimeError(msg)
+
+    def write_file(text: str) -> None:
+        """write the text to a file in the tmp_path directory
+
+        Args:
+            text (str): The text to write to the file.
+
+        Returns:
+            None
+        """
+        with open(os.path.join(tmp_path, tmp_file), "w", encoding="utf-8") as f:
+            f.write(text)
+
+    kwargs["model_id"] = "gpt-4.1-mini"
+    env_check = validate_environment(kwargs["model_id"])
+    if not env_check["keys_in_environment"]:
+        pytest.skip(f"{env_check['missing_keys']} needed for {agent_framework}")
+
+    model_args: dict[str, Any] = (
+        {"parallel_tool_calls": False}
+        if agent_framework not in [AgentFramework.AGNO, AgentFramework.LLAMA_INDEX]
+        else {}
+    )
+    model_args["temperature"] = 0.0
+    tools = [
+        write_file,
+        MCPStdio(
+            command="uvx",
+            args=["mcp-server-time", "--local-timezone=America/New_York"],
+            tools=[
+                "get_current_time",
+            ],
+        ),
+    ]
+    agent_config = AgentConfig(
+        tools=tools,  # type: ignore[arg-type]
+        instructions="Search the web to answer",
+        model_args=model_args,
+        **kwargs,  # type: ignore[arg-type]
+    )
+    agent_server = await AnyAgent.create_async(agent_framework, agent_config, tracing=TracingConfig())
+    logger.info(f"Agent created: {agent_server}")
+    await agent_server.serve(
+        serving_config=ServingConfig(port=5555,endpoint="/test_agent")
+    )
+    logger.info(f"Agent serving")
+
+    await asyncio.sleep(300)
```

Comment:
Adding sleeps makes me wary. Would be better to have some sort of health check to ping until ready, with a timeout.

Reply:
Yes, that's troubleshooting 😅 I keep it around during monkey testing to be able to ping it externally with curl.

Reply:
no sleep any more, except to wait for the uvicorn to start up and serve